### PR TITLE
Clean up the FitExtent control.

### PR DESCRIPTION
**Pull request by mhaertwig**

### CODE REVIEW

File: src/components/FitExtent/FitExtent.js [hunk 1]
```diff
@@ -33,25 +34,30 @@ const propTypes = {
...
+    <div
       className={className}
+      role="button"
       title={title}
-      onClick={() => {
-        map.getView().cancelAnimations();
-        map.getView().fit(extent, map.getSize());
-      }}
+      tabIndex="0"
+      onClick={() => fit()}
```
Comment (friedjoff) [inline src/components/FitExtent/FitExtent.js]: Does this work?

```suggestion
      onClick={fit}
```

Comment (mhaertwig) [reply src/components/FitExtent/FitExtent.js]: Of course, thanks :-)

File: src/components/FitExtent/FitExtent.js [hunk 2]
```diff
@@ -21,37 +21,37 @@ const propTypes = {
...
+      className="rs-fit-extent"
+      role="button"
       title={title}
-      onClick={() => {
-        map.getView().cancelAnimations();
-        map.getView().fit(extent, map.getSize());
-      }}
+      tabIndex="0"
+      onClick={fit}
+      onKeyPress={e => e.which === 13 && fit()}
```
Comment (oterral) [inline src/components/FitExtent/FitExtent.js]: you can also use `useCallBack` for key press

File: src/components/FitExtent/FitExtent.js [hunk 3]
```diff
@@ -21,37 +21,37 @@ const propTypes = {
...
-
-  /**
-   * Children content of the fitExtent button.
-   */
-  children: PropTypes.node,
+  children: PropTypes.node.isRequired,
 };
 
 const defaultProps = {
   title: 'Fit Extent',
```
Comment (oterral) [inline src/components/FitExtent/FitExtent.js]: Title can be removefrom props definition . use it like the className

File: src/components/FitExtent/FitExtent.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import PropTypes from 'prop-types';
-import { FaExpand } from 'react-icons/fa';
 import OLMap from 'ol/Map';
-import Button from '../Button';
+
+import './FitExtent.scss';
```
Comment (oterral) [inline src/components/FitExtent/FitExtent.js]: scss are loaded separately in the theme folder .

Comment (oterral) [reply src/components/FitExtent/FitExtent.js]: Scss moved to the example

File: src/components/FitExtent/FitExtent.js (hunk 2 above)
Comment (oterral) [reply src/components/FitExtent/FitExtent.js]: fixed

File: src/components/FitExtent/FitExtent.js (hunk 3 above)
Comment (oterral) [reply src/components/FitExtent/FitExtent.js]: fixed

**Pull request merged**